demoapp/demo_app_functions.py

```diff
@@ -450,4 +512,642 @@ def update_csv(bill_num: str, title: str, summarized_bill: str, category: str, t
+
+
+                    INSTRUCTIONS: 
+                    1. Choose just 2 categories from the list above.
+                    2. Do not provide explanations for the category choices.
+                    3. Do not output categories not listed above.
+                    4. Do not modify or paraphrase the category names, choose directly from the list provided.
+                    5. Respond with # separated categories
+
+                    Categories: """
+    else: 
+        query = """ 
+                Your job is to classify the bill according to the list of categories below. 
+                Choose the closest relevant category and do not output categories outside of this list. 
+                Please be politically neutral, keeping in mind that readers may have various ideological perspectives. 
+                Use the information from specified chapters and sections of the Mass General Laws to categorize the bill.
+
+                List of Categories:
+                {categories}
+
+                Note that the bill refers to specific existing chapters and sections of the Mass General Laws. Use the corresponding names of Mass General Law chapter and sections for constructing your summary.\n
+
+                The bill title is: \"{title}"\
+
+                The bill text is: \"{context}"\
+                
+                The relevant committee information: \"{committee_info}"\
+
+                The relevant section names are: \"{names}"\
+
+                The relevant section text is: \"{mgl_sections}"
+
+                INSTRUCTIONS: 
+                1. Choose just 2 categories from the list above.
+                2. Do not provide explanations for the category choices.
+                3. Do not output categories not listed above.
+                4. Do not modify or paraphrase the category names, choose directly from the list provided.
+                5. Respond with # separated categories
+
+                Categories: """
+        bill_details.invoke_dict = prepare_invoke_dict(bill_details)
+        bill_details.invoke_dict['categories'] = getattr(bill_details, 'categories', new_categories_for_bill_list)
+
+    return query
+
+def get_query_for_tagging(bill_details: BillDetails, category_tags: list, llm_call_type: str) -> str: 
+
+    """
+        
+    This functions prepares a prompt based on the call type (small: No use of RAG, large: Use RAG) for bill tagging
+
+    Args: 
+        bill_details (BillDetails): Object containing information about the bill - bill_text, bill_title, mgl_ref, commottee_info, mgl_names
+        category_tags (List): List of tags that the model has to filter from. 
+        llm_call_type (str): This argument can take 2 values ("small": No use of RAG, "large": Use RAG) 
+
+    Returns: 
+        str: Query that includes task, context and instructions on bill tagging. 
+
+    """
+
+    if llm_call_type == 'large':
+        query = f""" 
```

Comment:
It makes me really nervous how we hardcode so many versions of effectively the same query... I wish this was more templated. I proposed we at least add a TODO explaining this in a previous comment.

Reply:
I have created a new file for prompts only. 
Now it imports all the prompts and formats here in these functions